src/site/content/en/blog/granular-chunking-nextjs/index.md

```diff
@@ -0,0 +1,274 @@
+    <caption>JavaScript size reductions - across all routes (compressed)</caption>
+  </table>
+</div>
+
+The final version was shipped by default in [version 9.2](https://nextjs.org/blog/next-9-2).
+
+## Gatsby
+
+[Gatsby](https://www.gatsbyjs.org/) used to follow the same approach of utilizing a usage-based
+heuristic for defining common modules:
+
+```javascript/13
+config.optimization = {
+  //...
+  splitChunks: {
+    name: false,
+    chunks: `all`,
+    cacheGroups: {
+      default: false,
+      vendors: false,
+      commons: {
+        name: `commons`,
+        chunks: `all`,
+        // if a chunk is used more than half the components count,
+        // we can assume it's pretty global
+        minChunks: componentsCount > 2 ? componentsCount * 0.5 : 2,
+      },
+      react: {
+        name: `commons`,
+        chunks: `all`,
+        test: /[\\/]node_modules[\\/](react|react-dom|scheduler)[\\/]/,
+      },
+```
+
+By optimizing their webpack configuration to adopt a similar granular chunking strategy, they also
+noticed sizeable JavaScript reductions in many large sites:
+
+<div class="w-table-wrapper">
+  <table>
+    <thead>
+      <tr>
+        <th>Website</th>
+        <th>Total JS Change</th>
+        <th>% Difference</th>
+      </tr>
+    </thead>
+    <tbody>
+      <tr>
+        <td><a href="https://www.gatsbyjs.org/">https://www.gatsbyjs.org/</a></td>
+        <td>-680 KB</td>
+        <td>-22%</td>
+      </tr>
+      <tr>
+        <td><a href="https://www.thirdandgrove.com/">https://www.thirdandgrove.com/</a></td>
+        <td>-390 KB</td>
+        <td>-25%</td>
+      </tr>
+      <tr>
+        <td><a href="https://ghost.org/">https://ghost.org/</a></td>
+        <td>-1.1 MB</td>
+        <td>-35%</td>
+      </tr>
+      <tr>
+        <td><a href="https://reactjs.org/">https://reactjs.org/</a></td>
+        <td>-80 Kb</td>
+        <td>-8%</td>
+      </tr>
+    </tbody>
+    <caption>JavaScript size reductions - across all routes (compressed)</caption>
+  </table>
+</div>
+
+Take a look at the [PR](https://github.com/gatsbyjs/gatsby/pull/22253) to understand how they
+implemented this logic into their webpack configuration, which is shipped by default in v2.20.7.
+
+## Conclusion
+
+It's important to mention that the concept of shipping granular chunks is not specific to Next.js,
+Gatsby or even webpack. Everyone should consider improving their application's chunking strategy if
+it follows a large "commons" bundle approach, regardless of the framework or module bundler used.
+
++   If you would like to see the same chunking optimizations applied to a vanilla React application,
+    take a look at this [sample React
+    app](https://glitch.com/edit/#!/webpack-granular-split-chunks?path=webpack.config.js). It uses a
+    simplified version of the granular chunking strategy and can help you start applying the same
+    sort of logic to your site.
++   If you use Rollup instead of webpack, take a look at the
```

Comment:
Rollup chunks granularly by default, might be good to mention that here and then just refer folks to `manualChunks` if they want to impose set criteria atop that (max size, etc).

Reply:
Updated!